edweiss-app/app/(app)/courses/[id]/index.tsx

```diff
@@ -154,42 +145,34 @@ const CoursePage: ApplicationRoute = () => {
 				{upcomingAssignments.length > 0
 					?
 					upcomingAssignments.map((assignment) => (
-						<TView key={assignment.name} testID={testIDs.assignemtView}>
-							<AssignmentDisplay item={assignment} index={upcomingAssignments.indexOf(assignment)} isSwipeable={true} />
-						</TView>
+						<AssignmentDisplay item={assignment} index={upcomingAssignments.indexOf(assignment)} isSwipeable={true} key={assignment.name} />
 					))
 					:
 					<TText size={16} testID={testIDs.noAssignmentDue} >{t('course:no_assignment_due')}</TText>
 				}
 
 				{/* Bouton vers les Passed Assignments */}
-				<TTouchableOpacity onPress={() =>
-					router.push({
-						pathname: `/courses/[id]/archive`, // Forcer la chaîne à être interprétée comme littérale
-						params: { id: course.id, extraInfo: JSON.stringify(previousAssignments) }
-					})
-				}>
+				<TTouchableOpacity testID={testIDs.previousAssignmentTouchable} onPress={() => router.push({ pathname: `/courses/[id]/archive`, params: { id: course.id, extraInfo: JSON.stringify(previousAssignments) } })}>
```

Comment:
The `extraInfo` route parameter has a confusing name as it gives no information on what it expects, maybe there is a better way to name it ?